File: Week_2/Day_1/FirstUniqueCharacter.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq = {}

        # Count frequency
        for ch in s:
            freq[ch] = freq.get(ch, 0) + 1

        # Find first non-repeating character
        for i in range(len(s)):
            if freq[s[i]] == 1:
                return i

        return -1

File: Week_2/Day_1/FirstUniqueCharacter.java
import java.util.*;

class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character, Integer> freq = new HashMap<>();

        // Count frequency
        for (char ch : s.toCharArray()) {
            freq.put(ch, freq.getOrDefault(ch, 0) + 1);
        }

        // Find first non-repeating character
        for (int i = 0; i < s.length(); i++) {
            if (freq.get(s.charAt(i)) == 1) {
                return i;
            }
        }

        return -1;
    }
}